### DISCUSSION TITLE
Router cancellation

### DISCUSSION THREAD

**Discussion by tomonari-t** [vercel/next.js#11940] [Help] [answered]
In my application, I want to prevent user page transitions at form page.

Now I'm able to prevent page transitions that  user clicking `<Link>` or `Router.push('')`.
But it is not possible that preventing page transitions by clicking browser back button or refresh button.

It's very frustrating to lose data while filling out a form. And it often happens.

I'd prefer an official API for handling situations like confirming navigation.

How about everyone?

## Related Issue

[#2476](https://github.com/zeit/next.js/issues/2476)

**Comment by StarpTech** [comment, marked as answer]
Can be elegantly solved by https://github.com/zeit/next.js/discussions/11822

**Reply by tomonari-t** [reply]
@StarpTech 
Awesome !
